### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Daniil-Pankieiev**

### CODE REVIEW

File: app/clien.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+from typing import Type
+
+
+class Client:
```
Comment (LiudmylaKulinchenko) [inline app/clien.py:4]: Use Customer name, don't change it, you have an example inside a task description

File: app/data.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        name = client_data["name"]
+        client = Client(
+            name=name,
+            product_cart=client_data["product_cart"],
+            location=client_data["location"],
+            money=client_data["money"],
+            car=client_data["car"]
+        )
+        clients[name] = client
+    return clients
```
Comment (LiudmylaKulinchenko) [inline app/data.py:20]: Use list comprehension and unpacking (using `**`)
Hint: You can place Car creation inside Customer init

File: app/data.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    shops = {}
+    for shop_data in data["shops"]:
+        name = shop_data["name"]
+        shop = Shop(
+            name=name,
+            location=shop_data["location"],
+            products=shop_data["products"]
+        )
+        shops[name] = shop
+    return shops
```
Comment (LiudmylaKulinchenko) [inline app/data.py:33]: Use list comprehension and unpacking for shops creation

File: app/main.py [hunk 4]
```diff
@@ -1,3 +1,54 @@
...
+            client_info.shopping_costs.items(), key=lambda x: x[1]
+        )
+
+        if client_info.money < min_shop_cost:
+            print(f"{client_name} doesn't have enough money to "
+                  f"make a purchase in any shop")
+            return
+
+        print(f"{client_name} rides to {min_shop_name}")
+        print("")
```
Comment (LiudmylaKulinchenko) [inline app/main.py:30]: You can use print once and `\n` inside a string. Fix everywhere

**Review by LiudmylaKulinchenko — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 4 above)
Comment (Daniil-Pankieiev) [reply app/main.py]: Fixed

**Review by fedorov-s-od — APPROVED** [review]

**Review by LiudmylaKulinchenko — APPROVED** [review]
Good job 😸

**Pull request closed without merging**